### PR TITLE
usb_dc_stm32: ameliorate PMA leak on reset.

### PR DESCRIPTION
**Pull request by jmgao**
Previously, endpoint configuration would reserve memory in the packet
memory area which would never be reclaimed. After this patch, the leak
still exists on SET_CONFIGURATION/SET_INTERFACE, but we will at least
reclaim the memory upon reset, so disconnect/reconnect gets us back
into a working state.

Relates to #23178.

Signed-off-by: Josh Gao <josh@jmgao.dev>

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
I suggest to add a check if an endpoint is already configured or not to [usb_dc_ep_configure()](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/usb/device/usb_dc_stm32.c#L551) and fix the bug.

**Comment by carlescufi** [comment]
@loicpoulain @ydamigos could you take a look at this PR?

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -545,21 +546,25 @@ int usb_dc_ep_configure(const struct usb_dc_ep_cfg_data * const ep_cfg)
...
 
 	if (!ep_state) {
 		return -EINVAL;
 	}
 
 #ifdef USB
-	if (DT_USB_RAM_SIZE <=
-	    (usb_dc_stm32_state.pma_offset + ep_cfg->ep_mps)) {
-		return -EINVAL;
+	if (ep_cfg->ep_mps > ep_state->ep_pma_buf_len) {
```
Comment (ydamigos) [inline drivers/usb/device/usb_dc_stm32.c]: Why do we need `ep_state->ep_pma_buf_len`? Isn't enough to check `if (ep_cfg->ep_mps > ep_state->ep_mps)`?

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: drivers/usb/device/usb_dc_stm32.c (hunk 1 above)
Comment (ydamigos) [reply drivers/usb/device/usb_dc_stm32.c]: Ignore my comment. We need it because `ep_cfg->ep_mps` may be less than `ep_state->ep_mps`.

**Comment by ydamigos** [comment]
I am approving the patch as it improves the situation we have right now.

**Review by ydamigos — APPROVED** [review]

**Pull request merged**